spec.bs

```diff
@@ -3808,32 +3863,65 @@ The <dfn for=Navigator method>getInterestGroupAdAuctionData(|configIDL|)</dfn> m
   To <dfn>parse and verify ad auction data config</dfn> given an
   {{AdAuctionDataConfig}} |configIDL| and [=origin=] |top_level_origin|:
 
-  1. Let |seller| be the result of running [=parse an https origin=] on
-       |configIDL|["{{AdAuctionDataConfig/seller}}"].
-  1. Let |coordinator| be the result of running [=parse an https origin=] on
-       |configIDL|["{{AdAuctionDataConfig/coordinatorOrigin}}"].
-  1. If |seller| or |coordinator| are failure, then [=exception/throw=] a {{TypeError}}.
-  1. Let |config| be a new [=auction data config=] with the following [=struct/items=]:
-    : [=auction data config/publisher=]
-    :: |top_level_origin|
-    : [=auction data config/seller=]
-    :: |seller|
-    : [=auction data config/coordinator=]
-    :: |coordinator|
-    : [=auction data config/request size=]
-    :: |configIDL|["{{AdAuctionDataConfig/requestSize}}"] if it [=map/exists=], null otherwise
-    : [=auction data config/per buyer config=]
-    :: The result of running [=parse per buyer auction data configs=] on
-      |configIDL|["{{AdAuctionDataConfig/perBuyerConfig}}"]
-  1. If |config|'s [=auction data config/per buyer config=] [=map/is not empty=]
-     and |config|'s [=auction data config/request size=] is null:
-     1. Let |requestSize| be 0.
-     1. [=list/For each=] |buyerConfig| of |config|'s [=auction data config/per buyer config=]'s [=map/values=]:
+  1. Let |configs| be a new empty [=list=] of [=auction data configs=].
+  1. Let |per buyer configs| be the result of running [=parse per buyer auction data configs=] on
+      |configIDL|["{{AdAuctionDataConfig/perBuyerConfig}}"].
+  1. Let |requestSize| be |configIDL|["{{AdAuctionDataConfig/requestSize}}"] if it [=map/exists=], null otherwise.
+  1. If |per buyer configs| [=map/is not empty=] and |requestSize| is null:
+     1. Set |requestSize| to 0.
+     1. [=list/For each=] |buyerConfig| of |per buyer configs|'s [=map/values=]:
         1. If |buyerConfig|'s [=auction data buyer config/size=] is null,
            then [=exception/throw=] a {{TypeError}}.
         1. Set |requestSize| to |requestSize| + |buyerConfig|'s [=auction data buyer config/size=].
-    1. Set |config|'s [=auction data config/request size=] to |requestSize|.
-  1. Return |config|.
+  1. If |configIDL|["{{AdAuctionDataConfig/seller}}"] [=map/exists=]:
+    1. Let |seller| be the result of running [=parse an https origin=] on
+       |configIDL|["{{AdAuctionDataConfig/seller}}"].
+    1. If |configIDL|["{{AdAuctionDataConfig/coordinatorOrigin}}"] [=map/exists=]:
+      1. Let |coordinator| be the result of running [=parse an https origin=] on
+        |configIDL|["{{AdAuctionDataConfig/coordinatorOrigin}}"].
+    1. Otherwise
+      1. Let |coordinator| be the [=user agent=]'s default coordinator.

```

Comment:
let's say what coordinator it refers to, like "bidding and auction coordinator"? The spec has several different totally unrelated coordinators now.

Reply:
Done.